examples/drivenet/drivenet_train.ipynb

```diff
@@ -0,0 +1,321 @@
+    "\n",
+    "In our setting, we want to ensure **our DriveNet can recover if it ends up slightly off the midlane it is following**.\n",
+    "\n",
+    "To this end, we can enrich the training set with **online trajectory perturbations**. These perturbations are kinematically feasible and affect both starting angle and position. A new ground truth trajectory is then generated to link this new starting point with the original trajectory end point. These starting point will be slightly rotated and off the original midlane, and the new trajectory will teach the model how to recover from this situation.\n",
+    "\n",
+    "![perturbation](../../images/drivenet/perturb.svg)\n",
+    "\n",
+    "\n",
+    "In the following cell, we load the training data and leverage L5Kit to add these perturbations to our training set.\n",
+    "We also plot the same example with and without perturbation. During training, our model will see also those examples and learn how to recover from positional and angular offsets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perturb_prob = cfg[\"train_data_loader\"][\"perturb_probability\"]\n",
+    "\n",
+    "# rasterisation and perturbation\n",
+    "rasterizer = build_rasterizer(cfg, dm)\n",
+    "perturbation = AckermanPerturbation(\n",
+    "        random_offset_generator=GaussianRandomGenerator(mean=np.array([0.0, 0.0]), std=np.array([1.0, np.pi / 6])),\n",
+    "        perturb_prob=perturb_prob,\n",
+    "    )\n",
+    "\n",
+    "# ===== INIT DATASET\n",
+    "train_zarr = ChunkedDataset(dm.require(cfg[\"train_data_loader\"][\"key\"])).open()\n",
+    "train_dataset = EgoDataset(cfg, train_zarr, rasterizer, perturbation)\n",
+    "\n",
+    "# plot same example with and without perturbation\n",
+    "for perturbation_value in [1, 0]:\n",
+    "    perturbation.perturb_prob = perturbation_value\n",
+    "\n",
+    "    data_ego = train_dataset[0]\n",
+    "    im_ego = rasterizer.to_rgb(data_ego[\"image\"].transpose(1, 2, 0))\n",
+    "    target_positions = transform_points(data_ego[\"target_positions\"], data_ego[\"raster_from_agent\"])\n",
+    "    draw_trajectory(im_ego, target_positions, TARGET_POINTS_COLOR)\n",
+    "    plt.imshow(im_ego[::-1])\n",
+    "    plt.axis('off')\n",
+    "    plt.show()\n",
+    "\n",
+    "# before leaving, ensure perturb_prob is correct\n",
+    "perturbation.perturb_prob = perturb_prob"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model\n",
+    "L5Kit provides a model file for DriveNet. The backbone is a ResNetX (either 18 or 50) model pre-trained on ImageNet. Here, we describe the main inputs and outputs of the model; for a full description please check [the class definition](https://github.com/lyft/l5kit/blob/1d077fdbc8656057516bee2bb6cc815bc6868d29/l5kit/l5kit/drivenet/model.py#L9).\n",
+    "\n",
+    "#### Inputs\n",
```

Comment:
Maybe link / mention agent prediction here, mention that inputs etc are very similar?